pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -754,15 +754,18 @@ type OpenStackSDConfig struct {
 	Region string `json:"region"`
 	// IdentityEndpoint specifies the HTTP endpoint that is required to work with
 	// the Identity API of the appropriate version.
+	// +kubebuilder:validation:Pattern:="^http(s)?://.+$"
```

Comment:
I am not very good with regex but I tested this in https://regex101.com and it throws error.
Also I think it should `^http(s)?:\/\/.+$`, so adding the testcase in e2e would help to verify this